lib/TPP/ConvertTppToXsmm.cpp

```diff
@@ -285,6 +286,24 @@ struct ConvertTppFusedBrgemmOp : public OpRewritePattern<tpp::FusedBrgemmOp> {
            "tpp.fused_brgemm expects buffer semantics");
 
     Location loc = brgemmOp.getLoc();
+
+    // Current limitation in LIBXSMM.
+    // See: https://github.com/libxsmm/libxsmm/issues/766
+    // Split into separate operations if bcast_col_in0 is not present when add
+    // is fused.
+    // TODO: remove the split once LIBXSMM is fixed.
+    auto isBiasAdd = brgemmOp.getBinaryKind() == tpp::FusedBinaryOpKind::ADD;
+    auto binaryFlag = getBinaryFlags(rewriter, brgemmOp)[0]
+                          .cast<xsmm::BinaryFlagsAttr>()
+                          .getValue();
+    auto isBitSet = static_cast<uint64_t>(binaryFlag) &
+                    static_cast<uint64_t>(xsmm::BinaryFlags::BCAST_COL_IN_0);
+    if (isBiasAdd && !isBitSet) {
+      (void)tpp::utils::splitFusedOp(brgemmOp, rewriter);
```

Comment:
You're adding new ops with the rewriter without setting the insertion point. Is that guaranteed to be at the right place?